client/src/pages/registration/registationForms/doctorRegister/DoctorRegister.jsx

```diff
@@ -21,12 +29,205 @@ function DoctorRegister() {
+              .then((r) => r.blob())
+              .then(
+                (blobFile) =>
+                  new File(
+                    [blobFile],
+                    new Date().getTime().toString() +
+                      newUserUID +
+                      `${imgNum++}`,
+                    { type: blobFile.type }
+                  )
+              );
+            const uploadImgRef = ref(storage, `images/${imgObj.name}`);
+            await uploadBytes(uploadImgRef, imgObj);
+            const getDownloadableURL = await getDownloadURL(
+              ref(storage, `images/${imgObj.name}`)
+            );
+            setURLs.push(getDownloadableURL);
+          })
+        );
+        await setDoc(doc(db, "clinicImages", newUserUID), {
+          clinicImgURLs: setURLs,
+        });
+        setIsUserCreated(true);
+      } catch (e) {
+        console.log(e);
+        if (e.code === "auth/email-already-in-use") {
+          const oldErrorList = errorList;
+          oldErrorList["doctorEmail"] = ["Email ID already in use"];
+          setErrorList(oldErrorList);
+        } else {
+          throw e;
+        }
+      }
+      setLoading(false);
+    }
+  };
+
+  const ValidateImageHandler = (e) => {
+    const images = e.target.files;
+    const checkType = new Set([".jpg", ".jpeg", ".bmp", ".gif", ".png"]);
+    let typeErrorMsg = [];
+    if (images.length + displayImage.length > 5) {
+      typeErrorMsg.push("Cannot upload more than 5 images");
+      setErrorList({ ...errorList, clinicPictures: typeErrorMsg });
+    } else {
+      Object.values(images).forEach((img) => {
+        const fileFormat = "." + img.type.split("/").at(-1);
+        if (!checkType.has(fileFormat)) {
+          typeErrorMsg.push(`Incorrect File type of ${img.name}`);
+          setErrorList({ ...errorList, clinicPictures: typeErrorMsg });
+        } else if (img.size > 3000000) {
+          typeErrorMsg.push(
+            `${img.name} is too large. File size should be less than 3MBs`
+          );
+          setErrorList({ ...errorList, clinicPictures: typeErrorMsg });
+        } else {
+          setErrorList({ ...errorList, clinicPictures: [] });
+        }
+      });
+    }
+    if (typeErrorMsg.length === 0) {
+      let displayImg = [...displayImage];
+      Object.values(images).forEach((img) => {
+        displayImg.push(URL.createObjectURL(img));
+      });
+      setDisplayImage(displayImg);
+    }
   };
+
+  useEffect(() => {
+    async function fetchClinicOptions() {
+      try {
+        let retrievedData = await getDocs(collection(db, "formInputs"));
+        retrievedData.forEach((doc) => {
+          setTreatmentOptions(doc.data().treatments);
+          setSpecializationOptions(doc.data().specializations);
+        });
+      } catch (error) {
+        console.log(error);
+      }
+    }
+    fetchClinicOptions();
+  }, []);
+
+  useEffect(() => {
+    if (loading === false && isUserCreated) {
```

Comment:
can simplify to `!loading && isUserCreated`. In general, bools do not need to be checked for equality.